distribution/pom.xml

```diff
@@ -485,6 +485,36 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <!-- this profile adds/overrides few features of the 'apache-release'
+        profile in the parent pom. -->
+      <id>apache-release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.spotify</groupId>
+            <artifactId>dockerfile-maven-plugin</artifactId>
+            <version>1.4.3</version>
+            <executions>
+              <execution>
+                <id>docker-image</id>
+                <goals>
+                  <goal>build</goal>
+                  <goal>push</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <repository>drill/apache-drill-centos</repository>
```

Comment:
1. We might want to make this configurable? 
2. Why docker image build will be available only in apaceh-release profile? What if user want to create image and push it to its own repo?
3. `drill/apache-drill-centos` does committers have access to this repo to do the push?
4. https://github.com/parthchandra/drill/wiki/Drill-Release-Process should be updated with instructions about Docker.

Reply:
1. This is configurable - anyone can update it to any private or public Docker hub or any Container Registry. Do you have a different approach in mind?
2. I briefly touched upon this in the description. Since the ask was for an official Docker image for AD, it makes sense to build this once during the release. Anyone can download the Docker image at anytime, using a simple pull. Everything else remains the same. I also chose this approach to prevent Docker images from being generated everytime. 

3. Tim and I have. We intend to give access to the 'drill' Docker Hub Organization to all committers. Perhaps, starting with release managers for every release. It's a good central place to have all related docker images for Drill.

4. Sure.

Reply:
I suggest not keeping a separate repository based on os. Typically there is one repository and the os is included in the tag for the image.

```
drill/apache-drill:1.14-centos
drill/apache-drill:1.14-alpine
```

Reply:
We could move `drill/apache-drill-centos` into a maven property so that the user could override it from the command line.

Reply:
Sure. How about even simpler tag: 
drill/apache-drill:1.14.0
The default tag will be for CentOS. We can have additional tags like you mentioned:
drill/apache-drill:1.14.0-centos (alias for 1.14.0)
drill/apache-drill:1.14.0-ubuntu
drill/apache-drill:1.14.0-alpine 
etc..

Also sure, the Docker Hub repo can be moved into a maven property. Should we keep the drill docker hub repo as the default, or just make it a local repository by default instead.

Reply:
`Tim and I have. We intend to give access to the 'drill' Docker Hub Organization to all committers. Perhaps, starting with release managers for every release. It's a good central place to have all related docker images for Drill.`
I think you might consider giving access to several PMCs as well :)

Reply:
Sure! We'll need everyone to create an account on Docker Hub: https://hub.docker.com/u/drill/ is the repository. Once I have the usernames, I can add them to the 'drill' organization - that should give write access. How do you suggest I proceed with this? For now, I was planning on asking Boaz (current release manager) and anyone interested their usernames.